features/step_definitions/add_job_steps.js

```diff
@@ -0,0 +1,48 @@
+module.exports = function () {
+  const sampleJob = {
+    jobTitle: 'Test job',
+    employer: 'Test employer',
+  };
+
+  function addJob(job, accountId) {
+    return this.dashboardPage
+      .visit(accountId)
+      .then(() => this.dashboardPage.clickAddJobButton())
+      .then(() => this.addJobPage.fillJobApplication(Object.assign(sampleJob, job)));
+  }
+
+  function claimantJobTitle(claimant) {
+    return `job for ${claimant}`;
+  }
+
+  this.When(/^I add a job application$/, function () {
+    this.scenarioData.addedJob = sampleJob;
+    return addJob.call(this, sampleJob, this.scenarioData.accountIdentifier);
+  });
+
+  this.When(/^I add a job application without a title$/, function () {
+    return addJob.call(this, { jobTitle: '' }, this.scenarioData.accountIdentifier);
+  });
+
+  this.When(/^"([^"]*)" adds a job application$/, function (claimant) {
+    this.scenarioData.addClaimant(claimant);
+    return addJob.call(this, { jobTitle: claimantJobTitle(claimant) },
+      this.scenarioData.idFor(claimant));
+  });
+
+  this.Then(/^I should see a validation error$/, function () {
+    return this.expect(this.addJobPage.getValidationError()).to.equal('Job title is required');
+  });
+
+  // eslint-disable-next-line no-unused-vars
+  this.Then(/^"([^"]*)" should only see (his|her) job application$/, function (claimant, hisHer) {
```

Comment:
if we decided to keep this wording, then gender-neutral "only see their job application" simplifies the regex